mono/metadata/attach.c

```diff
@@ -577,8 +571,11 @@ receiver_thread (void *arg)
 			agent_args = decode_string_value (p, &p, p_end);
 
 			printf ("attach: Loading agent '%s'.\n", agent_name);
-			mono_attach_load_agent (mono_domain_get (), agent_name, agent_args, &exc);
-
+			ERROR_DECL (error);
+			mono_attach_load_agent (mono_domain_get (), agent_name, agent_args, error);
+			mono_error_cleanup (error);
+			g_free (agent_name);
```

Comment:
Cmd also leaks.
Protocol should probably null terminate to skip the alloc+copy+leak.